File: serverTCC/serverTCC/Models/HistoricoObjetivo.cs
using Newtonsoft.Json;
using System;

namespace serverTCC.Models
{
    public class HistoricoObjetivo
    {
        public int Id { get; set; }
        public DateTime DataFinal { get; set; }
        public decimal? ValorFinal { get; set; }
        [JsonIgnore]
        public int ObjetivoId { get; set; }
        [JsonIgnore]
        public Objetivo Objetivo { get; set; }
    }
}

File: client/www/js/models/historicoObjetivo.js
function HistoricoObjetivo(data) {
    "use strict";
    if (!data) data = {};

    this.id = data.id;
    this.dataFinal = data.dataFinal;
    this.valorFinal = data.valorFinal;
    this.objetivoId = data.objetivoId;
    if (data.objetivo)
        this.objetivo = new Objetivo(data.objetivo);
}

localEntities.register(HistoricoObjetivo, {
    id: {
        pk: true,
        type: 'number'
    },
    dataFinal: {
        type: 'Date'
    },
    valorFinal: {
        type: 'number'
    },
    objetivoId: {
        fk: true,
        type: 'number'
    },
    objetivo: {
        type: 'Objetivo',
        relationKey: 'objetivoId'
    }
});